### ISSUE TITLE
[Accessibility] Media alternative for time-based content

### ISSUE THREAD

**Issue by hostle83** [DSpace/dspace-angular#3807]
## Describe the bug
According to [WCAG Success Criterion 1.2.3](https://www.w3.org/TR/UNDERSTANDING-WCAG20/media-equiv-audio-desc.html), pre-recorded audio-only and video-only content must be accompanied by media alternatives for the visually impaired or the hearing impaired.

- Prerecorded Audio-only: An alternative for [time-based media](https://www.w3.org/WAI/WCAG21/Understanding/time-based-media) is provided that presents equivalent information for prerecorded audio-only content. This could be, for example, a link to a text transcript for an audio clip, 

- Prerecorded Video-only: Either an alternative for time-based media or an audio track is provided that presents equivalent information for prerecorded video-only content. For example, a silent movie includes an audio track which includes a description of the action in the video.

## Expected behavior
To satisfy this success criterion, either the system would need to automatically recognize that content is audio-only/video-only, or the user needs to be asked whether the uploaded content is audio-only/video-only. The software then needs to prompt the user to upload a media alternative.

At **Level A** in Success Criterion 1.2.3, authors have the **choice** of providing either an audio description or a full text alternative. If they wish to conform at **Level AA**, under Success Criterion 1.2.5 authors **must** provide an audio description - a requirement already met if they chose that alternative for 1.2.3, otherwise an additional requirement. At **Level AAA** under Success Criterion 1.2.8 they **must** provide an **extended** text **description**.

https://www.w3.org/TR/UNDERSTANDING-WCAG20/media-equiv-audio-desc-only.html

**Comment by atarix83** [comment]
@hostle83  @tdonohue 

This should be worked together with https://github.com/DSpace/dspace-angular/issues/3808 and this is our analysis on how we 'd resolve it: 

- Include a field in the submission form asking users to specify the type of media (e.g., audio-only, video-only, audio+video).

- Prompt for Alternative Content based on the media type selected:
  - For audio-only: prompt for a text transcript.
  - For video-only: prompt for a text description.
 
- Extend Metadata Schema adding metadata fields such as dc.description.transcript, dc.description.audiodescription or dc.description.textalternative
- Use Angular form logic to show/hide and validate fields based on the media type selected.
- Ensure transcripts and descriptions are clearly labeled and accessible in the item view.

estimation is about 24h

If it's ok we can work on both issues

**Comment by hostle83** [comment]
Thank you @atarix83! 

This looks like it's going to be a "larger" alteration, so I'm wondering if this needs to be discussed in dev meetings @tdonohue ? Essentially, we're talking not only about a programming, but also about a design decision here, right?

Just one comment regarding the second bullet point: 

- Prompt for Alternative Content based on the media type selected: For video-only: prompt for a text description. _I would not limit it to a text description - Ideally, the same video with **audio description**_ should be uploaded by the user

**Comment by tdonohue** [comment]
@atarix83 : I agree with @hostle83 that we may want to consider discussing some "mock ups" of your proposed changes in a future Developers Meeting.  This feature does sound like it may require some more significant changes to the Submission process (in the upload step), and it may be good to get early feedback on it before your team does much development.  That said the 24 hours sounds reasonable.

Just to make you aware, we are starting to approach the limit of our funds that have been set aside for `accessibility` tickets.  Therefore, if this ticket (or any other open `accessibility` ticket) end up going over budget, we'd need to be notified as soon as possible so that we can potentially secure more funding from Steering (if current accessibility funds are used up).

**Comment by tdonohue** [comment]
We discussed this ticket in today's [developers meeting](https://wiki.lyrasis.org/display/DSPACE/2025-06-05+DSpace+Developers+Meeting).  As a group, we decided that @atarix83 's [approach described above](https://github.com/DSpace/dspace-angular/issues/3807#issuecomment-2922747926) sounds like a good first step.  

There were lots of comments though on future work we should consider, which I'll move into a follow-up ticket.  That future work included:
* Seeing if we can use the file mimetype / bitstream format to determine the "type of media" in a more automated fashion.  This would require though fixing our reliance on the file extension to determine the mime type, as some file extensions (e.g. ".mp4") could be either video or audio.  Created at #4437 
* Eventually treating the transcript as a separate file to upload.  Transcripts can be very long, so they may not be ideal for metadata fields.  Additionally, if an Item has multiple audio files, then you'd need multiple transcripts (and a way to link each transcript to the correct audio file). Created at #4438
* Possibly also finding a way to improve our current [caption file upload process](https://wiki.lyrasis.org/display/DSDOC7x/User+Interface+Configuration#UserInterfaceConfiguration-Uploadingvideocaptioningfiles) (which is very manual).  Caption files and transcript files should likely be treated similarly, and we should prompt for both in the submission form, and "link" them up to the corresponding video/audio file. Included in #4438

~~I'll move these topics to separate ticket(s) so that they can be addressed separately.  Once those tickets are created, I'll update this comment with the links.~~  See linked tickets above.

**Comment by mwoodiupui** [comment]
I recalled that there has been past discussion of using something like JHOVE to automate file format discovery.  This would dovetail nicely with characterizing the media type of a submitted bitstream.

Where we *should* use metadata here is in linking related files, rather than depending on magic filename conventions.  This would be similar to linking versions.  Perhaps `dc.relation.isformatof` could back-link a transcript to the transcripted bitstream.  Too bad there is no `dc.relation.hasformat` to give the forward link.  There may be better concepts in some more specialized schema.  I will ask around.
